pyproject.toml

```diff
@@ -26,10 +26,44 @@ classifiers = [
     "Programming Language :: Python :: 3",
 ]
 dynamic = [
-  'version',
-  'readme',
-  'dependencies',
-  'optional-dependencies',
+    'version',
+    'readme',
+]
+dependencies = [
+    "boto3",
+    "boto3-utils",
+    "jsonpath-ng",
+    "python-json-logger",
+]
+
+[project.optional-dependencies]
+cli = [
+    "backoff",
+    "click>=8.2.0",
+    "click-option-group",
+]
+
+[dependency-groups]
+dev = [
+    "darglint",
+    "pytest",
+    "pytest-cov",
+    "pytest-mock",
+    "moto>=5",
+    "mypy",
+    "pip-tools",
+    "pre-commit",
+    "pre-commit-hooks",
+    "pymarkdownlnt",
+    "ruff",
+    "types-python-dateutil",
+]
+docs = [
+    "sphinx",
+    "myst_parser",
```

Comment:
I don't think we need this anymore. It was originally added to allow markdown README files for project tasks and workflows and whatnot to be consolidated into the docs when building for a specific project. Now that cirrus "projects" are no longer a thing, we don't need to support markdown documentation in sphinx, aside from what markdown-format docs sources we currently have. If we were to cleanup/reformat those to rst we could remove this dep and its configuration.

Note that I'm not suggesting adding more to this PR, I merely wanted to document this point somewhere.

Reply:
I'm going to defer on this until I make more progress on getting documentation update. Noted, and will implement then.